src/components/panel/panel.js

```diff
@@ -1636,6 +1685,91 @@ MdPanelRef.prototype._animateClose = function() {
   });
 };
 
+/**
+ * Registers a listener with the panel. The callback should return a promise,
+ * which will allow the action to continue when it gets resolved, or will
+ * prevent an action if it is rejected.
+ * @param {string} type Type of listener.
+ * @param {!angular.$q.Promise<!any>} callback Callback to be registered.
+ * @returns {!MdPanelRef}
+ */
+MdPanelRef.prototype.registerListener = function(type, callback) {
+  var error = null;
+
+  if (!angular.isString(type)) {
+    error = 'Listener type must be a string, instead got ' + typeof type;
```

Comment:
We did, although that's probably more appropriate once we have most of these PRs merged.